Bindings/Java/Matlab/Hopper_Device/BuildInteractiveHopperSolution.m

```diff
@@ -0,0 +1,217 @@
+                     'MillardTendonParams', MillardTendonParams, ...
+                     'maxIsometricForce', maxIsometricForce);
+if printSubcomponentAndOutputInfo
+    hopper.printSubcomponentInfo();
+end
+
+%% BUILD DEVICES
+devices = cell(0);
+deviceNames = cell(0);
+patellaWrap = cell(0);
+
+% Passive device
+if addPassiveDevice
+    passive = BuildDevice('deviceType','passive', ... 
+                          'springStiffness',springStiffness, ...
+                          'passivePatellaWrap',passivePatellaWrap);
+    devices{1,length(devices)+1} = passive;
+    deviceNames{1,length(deviceNames)+1} = 'device_passive';
+    patellaWrap{1,length(patellaWrap)+1} = passivePatellaWrap;  
+end
+
+% Active device
+if addActiveDevice
+    if isActivePropMyo
+        active = BuildDevice('deviceType','active', ...
+                             'isPropMyo',true, ...
+                             'gain',gain);
+    else
+        active = BuildDevice('deviceType','active', ...
+                             'isPropMyo',false, ... 
+                             'control',deviceControl, ...
+                             'maxTension',maxTension);
+    end
+    
+    devices{1,length(devices)+1} = active;
+    deviceNames{1,length(deviceNames)+1} = 'device_active';
+    patellaWrap{1,length(patellaWrap)+1} = activePatellaWrap;   
+end
+
+%% CONNECT DEVICES TO HOPPER
+for d = 1:length(devices)
+    
+    % Print the names of the device's subcomponents, and locate the
+    % subcomponents named 'anchorA' and 'anchorB'. Also, print the names of
+    % the hopper's subcomponents, and locate the two subcomponents named
+    % 'deviceAttach'.
+    device = devices{d};
+    if printSubcomponentAndOutputInfo
+        device.printSubcomponentInfo();
+    end
+    
+    % Get the 'anchor' joints in the device, and downcast them to the
+    % WeldJoint class. Get the 'deviceAttach' frames in the hopper
+    % model, and downcast them to the PhysicalFrame class.
+    anchorA = WeldJoint.safeDownCast(device.updComponent('anchorA'));
+    anchorB = WeldJoint.safeDownCast(device.updComponent('anchorB'));
+    thighAttach = PhysicalFrame.safeDownCast(...
+        hopper.getComponent('thigh/deviceAttach'));
+    shankAttach = PhysicalFrame.safeDownCast(...
+        hopper.getComponent('shank/deviceAttach'));
+    
+    % Connect the parent frame sockets of the device's anchor joints to the
+    % attachment frames on the hopper; attach anchorA to the thigh, and
+    % anchorB to the shank.
+    anchorA.connectSocket_parent_frame(thighAttach);
+    anchorB.connectSocket_parent_frame(shankAttach);
+    
+    % Add the device to the hopper model.
+    hopper.addComponent(device);
+    
+    % Configure the device to wrap over the patella.
+    if patellaWrap{d} && hopper.hasComponent([deviceNames{d}])
+        if strcmp(deviceNames{d},'device_passive')
+            % TODO: Change to downcast from PathSpring when PathSpring is
+            % working for passive device
```

Comment:
[HopperDevice](https://github.com/opensim-org/opensim-core/blob/tgcs2017/OpenSim/ExampleComponents/HopperDevice.h#L75) assumes a Force component that derives from PathActuator, which PathSpring doesn't.